project_generator/templates/makefile_armclang.tmpl

```diff
@@ -0,0 +1,33 @@
+{% extends "makefile.tmpl" %}
+
+{% block CC %}armclang{% endblock %}
+{% block CXX %}armclang{% endblock %}
+{% block AS %}armclang{% endblock %}
+{% block LD %}armlink{% endblock %}
+{% block AR %}armar{% endblock %}
+{% block CPP %}armclang{% endblock %}
+{% block OBJCOPY %}fromelf{% endblock %}
+{% block OBJDUMP %}fromelf{% endblock %}
+{% block SIZE %}fromelf{% endblock %}
+{% block NM %}fromelf{% endblock %}
+
+{% block TOBIN %}--bincombined{% endblock %}
+{% block TOHEX %}--i32combined{% endblock %}
+{% block objcopy_output %}--output{% endblock %}
+{% block objdump_output %}--output{% endblock %}
+{% block OBJDUMPFLAGS %}--text -acdyrz --interleave=source{% endblock %}
+{% block GENASMFLAGS %}-S -W{% endblock %}
+{% block SIZEFLAGS %}--text -z{% endblock %}
+{% block nm_output %}--output{% endblock %}
+{% block NMFLAGS %}--text -arz{% endblock %}
+
+{% block TARGET_EXE_EXT %}.axf{% endblock %}
+
+{% block COMMON_FLAGS %}--target=arm-arm-none-eabi -mcpu=$(CPU) -m$(INSTRUCTION_MODE){% endblock %}
+{% block CFLAGS %}$(COMMON_FLAGS) $(C_FLAGS) $(INC_DIRS_F) $(CC_SYMBOLS) -c -MMD{% endblock %}
+{% block CXXFLAGS %}$(COMMON_FLAGS) $(CXX_FLAGS) $(INC_DIRS_F) $(CC_SYMBOLS) -c -MMD{% endblock %}
+# Add -Wa,armasm,--show_cmdline to debug calls to armasm through armclang
```

Comment:
the comment is outdated as we use -masm=auto below?

Reply:
It's really more of project-generator developer comment. I struggled to find out exactly how `armclang` was actually invoking `armasm`. If you add `-Wa,armasm,--show_cmdline` to `ASFLAGS` it will display that nicely. Possibly a message to my future self 😄 .

Reply:
I still think we should use the gas syntax for `armclang` rather than `armasm`.

Reply:
I will remove the `-masm=auto` so it defaults to `-masm=gnu` and let projects set it through `as_flags` if they want to.

Reply:
Done in [96c9103]. I also removed `-x assembler-with-cpp`, which in a same way was an attempt to have things working out-of-the-box in more situation, but could potentially surprise users.

Reply:
Yay for `-masm`, but I'm not so sure about `assembler-with-cpp`. Having the assembler run through preprocessor is pretty innocuous, and a good default imo. We really need progen project config options, so we can have `assembler-with-cpp` enabled by default with user control from the yaml. But ultimately I'm fine with it since you _can_ enable it within the project.